src/Service/SF1500/AdresseFetchService.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace App\Service\SF1500;
+
+use App\Entity\SF1500\AdresseRegistrering;
+use App\Entity\SF1500\AdresseRegistreringEgenskab;
+use App\Exception\UnhandledException;
+use App\Service\SF1500Service;
+use Doctrine\ORM\EntityManagerInterface;
+use ItkDev\Serviceplatformen\SF1500\Adresse\ServiceType\_List;
+use ItkDev\Serviceplatformen\SF1500\Adresse\ServiceType\Soeg;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\EgenskabType;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\FiltreretOejebliksbilledeType;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\ListInputType;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\RegistreringType;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\RelationListeType;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\SoegInputType;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\SoegOutputType;
+use ItkDev\Serviceplatformen\SF1500\Adresse\StructType\TilstandListeType;
+use Psr\Log\LoggerAwareTrait;
+
+class AdresseFetchService implements FetchServiceInterface
+{
+    use LoggerAwareTrait;
+
+    public function __construct(private readonly EntityManagerInterface $entityManager, private readonly SF1500Service $sf1500Service)
+    {
+    }
+
+    public function fetch(int $pageSize, int $max): void
+    {
+        $total = 0;
+
+        while (true) {
+            $this->logger->debug(sprintf('Fetching adresse data, offset: %d , max: %d', $total, $max));
+            $this->logger->debug(sprintf('Memory used: %d ', memory_get_usage() / 1024 / 1024));
```

Comment:
This code is duplicated multiple times. Consider creating an abstract parent class for doing this, e.g.

```php
abstract class AbstractDataFetcher {
  public function fetchData(int $size, int $max = null): void {
    $total = 0;

    $this->preFetchData();

    while (true) {
      $chuckSize = $this->fetchChunk($offset, $size);
      if (false === $chunkSize) {
        break;
      }

      $total += $chunkSize;
    }
  }

  // Return false to break out of fetch loop
  abstract protected fetchChunk($offset, $size);

  protected preFetchData($offset, $size) {
    // Some fetchers need to do some initialization before starting the fetch loop.
  }
}
```

Reply:
Added `AbstractDataFetcher` to handle some of the code duplication.